src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/SignVerify.cs

```diff
@@ -37,7 +37,16 @@ public static void ExpectedSignature_SHA1_384()
                 0x11, 0xE0, 0x6E, 0xD2, 0x22, 0x75, 0xE7, 0x7C,
             };
 
-            ExpectSignature(expectedSignature, TestData.HelloBytes, "SHA1", TestData.RSA384Parameters);
+            try
+            {
+                ExpectSignature(expectedSignature, TestData.HelloBytes, "SHA1", TestData.RSA384Parameters);
+
+                Assert.True(RSAFactory.Supports384PrivateKey, "RSAFactory.Supports384PrivateKey");
+            }
+            catch (CryptographicException)
+            {
+                Assert.False(RSAFactory.Supports384PrivateKey, "RSAFactory.Supports384PrivateKey");
```

Comment:
Rather than asserting here, should we just propagate the exception if Supports384PrivateKey is true?